presto-hbase/src/main/java/com/facebook/presto/hbase/HBaseSplitManager.java

```diff
@@ -0,0 +1,114 @@
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import io.airlift.log.Logger;
+
+/**
+ * HBase specific implementation of {@link ConnectorSplitManager}.
+ */
+public class HBaseSplitManager
+        implements ConnectorSplitManager
+{
+    private static final Logger log = Logger.get(HBaseSplitManager.class);
+
+    private final String connectorId;
+    private final HBaseSimpleConsumerManager consumerManager;
+    private final Set<HostAddress> nodes;
+
+    @Inject
+    public HBaseSplitManager(
+            HBaseConnectorId connectorId,
+            HBaseConnectorConfig hbaseConnectorConfig,
+            HBaseSimpleConsumerManager consumerManager)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+        this.consumerManager = requireNonNull(consumerManager, "consumerManager is null");
+
+        requireNonNull(hbaseConnectorConfig, "hbaseConnectorConfig is null");
+        this.nodes = ImmutableSet.copyOf(hbaseConnectorConfig.getNodes());
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorTableLayoutHandle layout)
+    {
+    	ImmutableList.Builder<ConnectorSplit> splits = ImmutableList.builder();
+        
+    	try {
+	        HBaseTableHandle hbaseTableHandle = convertLayout(layout).getTable();
+	        TableName tableName = TableName.valueOf(hbaseTableHandle.getTopicName());
+	
+	        Connection connection = consumerManager.getConsumer(selectRandom(nodes));
+	        
+	        RegionLocator regionLocator=  connection.getRegionLocator(tableName);
+	        List<HRegionLocation> regionLocationList = regionLocator.getAllRegionLocations();
+	
+	
+	        
+	        for (HRegionLocation regionLocation : regionLocationList) {
+	        	HRegionInfo regionInfo = regionLocation.getRegionInfo();
+	            log.debug("Adding Region %s", regionInfo.getEncodedName());
+	            HBaseSplit split = new HBaseSplit(
+	                    connectorId,
+	                    hbaseTableHandle.getTopicName(),
+	                    hbaseTableHandle.getKeyDataFormat(),
+	                    hbaseTableHandle.getMessageDataFormat(),
+	                    regionInfo.getRegionId(),
+	                    regionInfo.getStartKey(),
+	                    regionInfo.getEndKey(),
+	                    HostAddress.fromParts(regionLocation.getHostname(), regionLocation.getPort())
+	                    );
+	
+	            splits.add(split);
+	        }
+    	} catch (IOException ex) {
+    		log.error(ex.getMessage(), ex);
+    	}
+        
+        return new FixedSplitSource(splits.build());
+    }
+
+    private static <T> T selectRandom(Iterable<T> iterable)
+    {
+        List<T> list = ImmutableList.copyOf(iterable);
```

Comment:
maybe store nodes as list, so you would not need to copy them each time you find splits